daras_ai_v2/base.py

```diff
@@ -2276,43 +2280,255 @@ def render_published_run_preview(self, published_run: PublishedRun):
+                                    with gui.link(
+                                        to=published_run.get_app_url(),
+                                    ):
+
+                                        gui.write(
+                                            f"#### {truncate_text_words(tb.h1_title, 50)}",
+                                        )
+                                with gui.div(
+                                    className="d-md-flex d-none align-items-center mb-1 ms-2",
+                                    style={"font-size": "0.9rem"},
+                                ):
+                                    for pill in pills:
+                                        pill()
+                            if published_run.notes:
+                                with gui.div(
+                                    className="d-none d-md-block container-margin-reset"
+                                ):
+                                    gui.caption(
+                                        truncate_text_words(
+                                            published_run.notes, max_desc_words
+                                        ),
+                                    )
+                                # mobile
+                                with gui.div(className="d-md-none"):
+                                    gui.caption(
+                                        truncate_text_words(
+                                            published_run.notes, max_desc_words_mobile
+                                        ),
+                                        style={"fontSize": "14px"},
+                                    )
+            # column 3
+            # web
+            if not hide_author_column:
+                with gui.div(
+                    className="d-none d-md-block col-12 col-md-2 flex-grow-1 mt-2"
+                ):
+                    (
+                        self.render_author(
+                            published_run.last_edited_by,
+                            image_size="24px",
+                            text_size="0.9rem",
+                            responsive=False,
+                            show_as_link=False,
+                        )
+                        if not use_workspace_author
+                        else self.render_workspace_author(
+                            published_run.workspace,
+                            image_size="24px",
+                            text_size="0.9rem",
+                            responsive=False,
+                            show_as_link=False,
+                        )
+                    )
+                    if version.change_notes:
+                        with gui.div(className="mt-2"):
+                            gui.caption(
+                                f"{icons.notes} {html.escape(truncate_text_words(version.change_notes, 20))}",
+                                unsafe_allow_html=True,
+                            )
```

Comment:
please break down chunks with their own responsibility into their own functions, so it can be repeated / easier to understand. e.g.

```python
_render_author_column(
    published_run=published_run,
    change_notes=change_notes,
    show_workspace=show_workspace,
)
```